atc/atccmd/command.go

```diff
@@ -1060,28 +1084,38 @@ func run(runner ifrit.Runner, onReady func(), onExit func()) ifrit.Runner {
 func (cmd *RunCommand) validate() error {
 	var errs *multierror.Error
 
-	tlsFlagCount := 0
-	if cmd.TLSBindPort != 0 {
-		tlsFlagCount++
-	}
-	if cmd.TLSCert != "" {
-		tlsFlagCount++
-	}
-	if cmd.TLSKey != "" {
-		tlsFlagCount++
-	}
-
-	if tlsFlagCount == 3 {
+	switch {
+	case cmd.TLSBindPort == 0:
+		if cmd.TLSCert != "" || cmd.TLSKey != "" || cmd.EnableAutocert {
+			errs = multierror.Append(
+				errs,
+				errors.New("must specify --tls-bind-port to use TLS"),
+			)
+		}
+	case cmd.EnableAutocert:
+		if cmd.TLSCert != "" || cmd.TLSKey != "" {
+			errs = multierror.Append(
+				errs,
+				errors.New("cannot enable autocert if --tls-cert or --tls-key are set"),
+			)
+		}
+		if cmd.TLSBindPort != 443 {
```

Comment:
Where does this requirement come from? :thinking: This may be problematic for setups that run `web` on a higher port (>1024) and use a load balancer to forward to it.

Reply:
This is probably a misunderstanding reading the autocerts Listeners which only listens on port 443. I have done some testing to make sure the ALPN challenge isn't restricted to having the external port on 443 so this check can be removed.